src/appcast.cpp

```diff
@@ -163,24 +194,40 @@ void XMLCALL OnStartElement(void *data, const char *name, const char **attrs)
         {
             if (!ctxt.items.empty())
             {
+                Appcast enclosure_item;
                 Appcast& item = ctxt.items.back();
+
                 for (int i = 0; attrs[i]; i += 2)
                 {
                     const char* name = attrs[i];
                     const char* value = attrs[i + 1];
 
                     if (strcmp(name, ATTR_URL) == 0)
-                        item.DownloadURL = value;
+                        enclosure_item.DownloadURL = value;
                     else if (strcmp(name, ATTR_VERSION) == 0)
-                        item.Version = value;
+                        enclosure_item.Version = value;
                     else if (strcmp(name, ATTR_SHORTVERSION) == 0)
-                        item.ShortVersionString = value;
+                        enclosure_item.ShortVersionString = value;
                     else if (strcmp(name, ATTR_DSASIGNATURE) == 0)
-                        item.DsaSignature = value;
+                        enclosure_item.DsaSignature = value;
                     else if (strcmp(name, ATTR_OS) == 0)
-                        item.Os = value;
+                        enclosure_item.Os = value;
                     else if (strcmp(name, ATTR_ARGUMENTS) == 0)
-                        item.InstallerArguments = value;
+                        enclosure_item.InstallerArguments = value;
+                }
+
+                // Set any enclosure data if no Os is set, otherwise just set
+                // data if it is suitable for windows.
+                // This might set the wrong operating system but accepts enclosures
+                // without Os information.
+                if (item.Os.empty() || is_suitable_windows_item(enclosure_item))
+                {
+                    item.Os = enclosure_item.Os;
+                    item.DownloadURL = enclosure_item.DownloadURL;
+                    item.Version = enclosure_item.Version;
+                    item.ShortVersionString = enclosure_item.ShortVersionString;
+                    item.DsaSignature = enclosure_item.DsaSignature;
+                    item.InstallerArguments = enclosure_item.InstallerArguments;
```

Comment:
This is error-prone and will *silently* break if any attribute is added to `Appcast`. If such copying is needed (it's not entirely clear to me why), it should be handled by having a copyable sub-object for the enclosure in the `Appcast` class.

Reply:
Hm, I would argue it is not more error prone than it was before: It would also silently break if more attributes are added in the original version.

It seemed handy to use Appcast but maybe individual strings or an extra structure for the attributes would make the code more readable.

The reason for the extra copy-step is to avoid directly overwriting data in Appcast without checking the data beforehand.

Initially I thought of introducing a structure for the enclosure data. But I could not find a clear documentation of the appcast.xml and it seems that it evolved over time and places for information changed. (e.g.: "version" can be a node and/or can be set in the enclosure tag?)
My conclusion was to not introduce any extra structure and use what is there to make minimal changes only.